conmon/conmon.c

```diff
@@ -1388,7 +1393,10 @@ int main(int argc, char *argv[])
 	 * Glib does not support SIGCHLD so use SIGUSR1 with the same semantic.  We will
          * catch SIGCHLD and raise(SIGUSR1) in the signal handler.
 	 */
-	g_unix_signal_add (SIGUSR1, on_sigusr1_cb, pid_to_handler);
+	g_unix_signal_add(SIGUSR1, on_sigusr1_cb, pid_to_handler);
+	signal(SIGTERM, on_sig_exit);
+	signal(SIGQUIT, on_sig_exit);
+	signal(SIGINT, on_sig_exit);
```

Comment:
I think we should move these before the fork to avoid a post-fork hole where they aren't installed ([handlers are inherited through a fork][1]).  Then condition the child-forwarding on a positive `container_pid`.

[1]: http://man7.org/linux/man-pages/man2/fork.2.html